src/parser/core/Module.ts

```diff
@@ -0,0 +1,273 @@
+import {cloneDeep} from 'lodash'
+import 'reflect-metadata'
+
+import {Event, Ability} from 'fflogs'
+import Parser from './Parser'
+
+export const DISPLAY_ORDER = {
+	TOP: 0,
+	DEFAULT: 50,
+	BOTTOM: 100,
+}
+
+export function dependency(target: Module, prop: string) {
+	const dependency = Reflect.getMetadata('design:type', target, prop)
+	const constructor = target.constructor as typeof Module
+
+	// Make sure we're not modifying every single module
+	if (constructor.dependencies === Module.dependencies) {
+		constructor.dependencies = []
+	}
+
+	// If the dep is Object, it's _probably_ from a JS file. Fall back to simple handling
+	if (dependency === Object) {
```

Comment:
Could also write warning in development with a suggested fix.

```js
import {About} from 'parser/modules/About'

/** @type {About} */
@dependency foo
```

Note that the decorator metadata needs to be able to access the concrete value of the constructor.

Reply:
I _hate_ comments having a concrete effect on code. They shouldn't. Ever. Ever ever ever.

So nah. I'm not going to be pushing for `@dependency` to be used in JS - it's only really there so TS gets proper typing for module deps. That check's just there so people _can_ use it in JS without breaking everything.